src/OneinchCallee.sol

```diff
@@ -0,0 +1,168 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+// Copyright (C) 2023 Dai Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.6.12;
+pragma experimental ABIEncoderV2;
+
+interface GemJoinLike {
+    function dec() external view returns (uint256);
+    function gem() external view returns (TokenLike);
+    function exit(address, uint256) external;
+}
+
+interface DaiJoinLike {
+    function dai() external view returns (TokenLike);
+    function join(address, uint256) external;
+}
+
+interface TokenLike {
+    function approve(address, uint256) external;
+    function transfer(address, uint256) external;
+    function balanceOf(address) external view returns (uint256);
+}
+
+interface CharterManagerLike {
+    function exit(address crop, address usr, uint256 val) external;
+}
+
+interface IAggregationExecutor {
+    function callBytes(address msgSender, bytes calldata data) external payable;
+}
+
+interface OneinchRouter {
+    struct SwapDescription {
+        TokenLike srcToken;
+        TokenLike dstToken;
+        address srcReceiver;
+        address dstReceiver;
+        uint256 amount;
+        uint256 minReturnAmount;
+        uint256 flags;
+    }
+
+    function swap(
+        IAggregationExecutor executor,
+        SwapDescription calldata desc,
+        bytes calldata permit,
+        bytes calldata data
+    ) external payable returns (uint256 returnAmount, uint256 spentAmount);
+}
+
+contract OneinchCallee {
+    DaiJoinLike public daiJoin;
+    TokenLike public dai;
```

Comment:
These should be immutable.

Reply:
Thanks! I understand that's a best practice, even if there is no methods to modify them. I was confused, since they are not marked as immutable in almost [every](https://github.com/makerdao/exchange-callees/blob/a52a88561d7edd3a2cc352d6030e62288f0899d2/src/PSMCallee.sol#L43-L44) [other](https://github.com/makerdao/exchange-callees/blob/a52a88561d7edd3a2cc352d6030e62288f0899d2/src/UniswapV3Callee.sol#L57-L59) callee in this repo 🙂

Reply:
Yeah I think some callees were written with an old solidity version before immutables were supported, and possibly some code was copied from those.